exercises/clock/src/example.c

```diff
@@ -15,17 +17,41 @@ static void normalize_clock(int *hour, int *minute)
    *minute %= 60;
 }
 
-void clock(time_text_t time_text, int hour, int minute)
+static void to_int(clock_t clock, int *hour, int *minute)
 {
+   sscanf(clock.text, CLOCK_FORMAT, hour, minute);
+}
+
+clock_t clock_create(int hour, int minute)
+{
+   clock_t clock = { {0} };
    normalize_clock(&hour, &minute);
-   sprintf(time_text, "%02d:%02d", hour, minute);
+   sprintf(clock.text, CLOCK_FORMAT, hour, minute);
+   return clock;
+}
+
+clock_t clock_add(clock_t clock, int minute_add)
+{
+   int hour = 0, minute = 0;
+   to_int(clock, &hour, &minute);
+   minute += minute_add;
+   return clock_create(hour, minute);
+}
+
+clock_t clock_subtract(clock_t clock, int minute_subtract)
+{
+   int hour = 0, minute = 0;
+   to_int(clock, &hour, &minute);
+   minute -= minute_subtract;
+   return clock_create(hour, minute);
 }
 
-void clock_add(time_text_t time_text, int minute_offset)
+bool clock_is_equal(clock_t a, clock_t b)
 {
-   int hour;
-   int minute;
-   sscanf(time_text, "%02d:%02d", &hour, &minute);
-   minute += minute_offset;
-   clock(time_text, hour, minute);
+   int ah = 0, am = 0, bh = 0, bm = 0;
+   to_int(a, &ah, &am);
+   to_int(b, &bh, &bm);
+   normalize_clock(&ah, &am);
+   normalize_clock(&bh, &bm);
+   return (ah == bh) && (am == bm);
```

Comment:
Can't you use strcmp here?

Reply:
Need to convert to `int` for the normalisation. E.g. `strcmp("24:00", "00:00")` does not return true, but for a clock it should. The canonical tests specifically test for these scenarios.